### PR TITLE
Adding like to cheeps

### LINKED ISSUE #241: As a user, I would like to react to cheeps

### PR DESCRIPTION
**Pull request by Alex-itu**
Added the ability to react to cheeps with an upvote and downvote input.

Resolves #241

### CODE REVIEW

**Review by Slug-Boi — CHANGES_REQUESTED** [review]
It looks like a solid implementation of adding reactions my only feedback which may be more of a question is whether reactions are like follows in the sense that they can't really exist without a cheep and thus maybe the reaction repo should instead be merged with the cheep repo. I propose that we chat about this Wednesday or Thursday. If this isn't the case then it's fully approved for merge!

**Comment by Alex-itu** [comment]
> It looks like a solid implementation of adding reactions my only feedback which may be more of a question is whether reactions are like follows in the sense that they can't really exist without a cheep and thus maybe the reaction repo should instead be merged with the cheep repo. I propose that we chat about this Wednesday or Thursday. If this isn't the case then it's fully approved for merge!

I see your point, but I think what we should do, is move followers out of Userrepo.  That way keep all functionalty for each data model separate from eachother. So let's make a new issue and branch for that.

**Review by Slug-Boi — APPROVED** [review]

**Review by JonasSkjodt — APPROVED** [review]
looks good, happy to see the cheeps have ids now. Make sure to clean up your comments youre not using and write some more comments explaining your code. There is some code which could perhaps reworked to work better, but perhaps we could do that together. See below.

**Review by August-Brandt — CHANGES_REQUESTED** [review]
Once requested changes are done then this looks good

**Comment by Haruyzal** [comment]
I might have been tinkering with this branch until 3 AM. I kindly ask you to pull AddingLikeToCheepsRefactoring and check out some of the suggestions. I'm aware I forgot to put in comments, so please feel free to write upon any uncertainties.

File: src/Chirp.Infrastructure/ChirpRepository/ReactionRepository.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    {
+        return await DbSetReaction.CountAsync(r => r.cheepId == id && r.reactionType.Equals("Upvote"));
+    }
+
+    public async Task<int> GetCheepsDownvoteCountsFromCheepID(int id) 
+    {
+        return await DbSetReaction.CountAsync(r => r.cheepId == id && r.reactionType.Equals("Downvote"));
+    }
+
+    public async Task ReactToCheep(ReactionDTO reactionDTO)
```
Comment (August-Brandt) [inline src/Chirp.Infrastructure/ChirpRepository/ReactionRepository.cs:58]: This can be simplefied after the changes to the data model

**Review by August-Brandt — CHANGES_REQUESTED** [review]
Just trying to see what "Start a review" does

File: src/Chirp.Infrastructure/ChirpRepository/ReactionRepository.cs (hunk 1 above)
Comment (August-Brandt) [inline src/Chirp.Infrastructure/ChirpRepository/ReactionRepository.cs:58]: This method can be simplified after the changes to data model

**Review by August-Brandt — CHANGES_REQUESTED** [review]
Still some changes to be made

**Comment by Haruyzal** [comment]
> I might have been tinkering with this branch until 3 AM. I kindly ask you to pull AddingLikeToCheepsRefactoring and check out some of the suggestions. I'm aware I forgot to put in comments, so please feel free to write upon any uncertainties.

It doth seemeth I forgot to push. Ignoer what I said. I'll remake the changes from memory or smth if y'all finish making changes and merging today.

**Review by August-Brandt — APPROVED** [review]
Look good now

**Pull request merged**